src/main/java/com/google/api/codegen/transformer/SurfaceNamer.java

```diff
@@ -559,18 +483,14 @@ public String getMockGrpcServiceImplName(Interface service) {
     return className(Name.upperCamel("Mock", service.getSimpleName(), "Impl"));
   }
 
-  /**
-   * The file name for an API service.
-   */
+  /** The file name for an API service. */
   public String getServiceFileName(Interface service, String packageName) {
     return getNotImplementedString("SurfaceNamer.getApiName");
   }
 
-  /**
-   * The name of the class that implements a particular proto interface.
-   */
-  public String getFullyQualifiedApiWrapperClassName(Interface interfaze, String packageName) {
-    return getNotImplementedString("SurfaceNamer.getFullyQuallifiedApiWrapperClassName");
+  /** The name of the class that implements a particular proto interface. */
+  public String getFullyQualifiedApiWrapperClassName(Interface service, String packageName) {
+    return packageName + "." + getApiWrapperClassName(service);
```

Comment:
This dot is not cross-language. The pattern in SurfaceNamer is to pass a language-agnostic holder of the text (e.g. in Name, NamePath) and pass those along to a method in NameFormatterDelegator/NameFormatter for the language-specific logic.